scripts/tools/linux_ip_namespace_setup.sh

```diff
@@ -0,0 +1,194 @@
+
+  # Give the host a known IPv6 addr and set the host side up
+  ip -6 addr add ${HOST_IPV6_ADDR}/64 dev ${HOST_SIDE_IF_NAME}
+  ip link set ${HOST_SIDE_IF_NAME} up
+
+  # Associate namespace IF with the namespace
+  ip link set ${NAMESPACE_SIDE_IF_NAME} netns ${NAMESPACE}
+
+  # Give the namespace IF an address (something nothing else is using) and set it up
+  echo "Adding address for namespace IF"
+  ip netns exec ${NAMESPACE} ip -6 addr add ${NAMESPACE_IPV6_ADDR}/64 dev ${NAMESPACE_SIDE_IF_NAME}
+  ip netns exec ${NAMESPACE} ip link set dev ${NAMESPACE_SIDE_IF_NAME} up
+
+  # Add a route to the namespace to go through the bridge
+  echo "Setting routes for namespace"
+  ip netns exec ${NAMESPACE} ip -6 route add default dev ${NAMESPACE_SIDE_IF_NAME}
+
+  echo "Setup complete."
+}
+
+function run_add_ipv4() {
+  # Give the namespace an IPv4 address
+  ip netns exec ${NAMESPACE} ip addr add ${NAMESPACE_ADDR}/24 dev ${NAMESPACE_SIDE_IF_NAME}
+
+  # Add a bridge, give it an address (something nothing else is using)
+  echo "Setting up bridge"
+  ip link add name ${BRIDGE_NAME} type bridge
+  ip -6 addr add ${BRIDGE_IPV6_ADDR}/64 dev ${BRIDGE_NAME}
+  ip addr add ${BRIDGE_ADDR}/24 brd + dev ${BRIDGE_NAME}
+
+  # For ipv6 and ipv4 to work together, need the bridge to ignore the ipv6 packets (DROP here means don't bridge)
+  ebtables-legacy -t broute -A BROUTING -p ipv6 -j DROP -i ${HOST_SIDE_IF_NAME}
+  ip link set ${BRIDGE_NAME} up
+
+  # Connect the host side to the bridge, so now we have bridge <-> host_side_if <-> namespace_if
+  echo "Connecting host virtual IF to bridge"
+  ip link set ${HOST_SIDE_IF_NAME} master ${BRIDGE_NAME}
+
+  #ip netns exec ${NAMESPACE} ip route add default via ${BRIDGE_ADDR} dev ${NAMESPACE_SIDE_IF_NAME}
+}
+
+function run_cmd() {
+  # Start the app in the namespace
+  echo "Running $1 in namespace."
+  ip netns exec ${NAMESPACE} $1
+}
+
+function run_cleanup() {
+  # Deleting the namespace will remove the namespace and peer'd interfaces to
+  have_ebtables_legacy=$1
+  ip netns delete ${NAMESPACE}
+  if ifconfig | grep ${BRIDGE_NAME}; then
+    if [ $have_ebtables_legacy = true ]; then
+      # Just try to drop the additional rule - it references our interface
+      # so if it's there, we added it.
+      ebtables-legacy -t broute -D BROUTING -p ipv6 -j DROP -i ${HOST_SIDE_IF_NAME} > /dev/null
+    fi
+    ip link delete dev ${BRIDGE_NAME} type bridge
+  fi
+}
+
+function help() {
+    echo "Usage: $file_name [ options ... ]"
+    echo ""
+
+    echo "This script is used to set up linux namespaces for Matter device testing."
+    echo "To use this script, set up a namespace then run the Matter device example"
+    echo "using the -r command. Run the controller in another terminal to send"
+    echo "commands to and from the device."
+    echo ""
+    echo "This script requires sudo for setup and requires access to ebtables-legacy"
+    echo "to set up dual ipv4/ipv6 namespaces. Defaults to ipv6 only."
+    echo ""
```

Comment:
Suggest adding a couple fully-worked examples for those not familiar, otherwise the likelihood of this script being used will be low.

Reply:
done.